android/java/org/chromium/chrome/browser/settings/BraveVpnPreferences.java

```diff
@@ -0,0 +1,448 @@
+                            .addTransportType(NetworkCapabilities.TRANSPORT_VPN)
+                            .removeCapability(NetworkCapabilities.NET_CAPABILITY_NOT_VPN)
+                            .build();
+            connectivityManager.registerNetworkCallback(networkRequest, mNetworkCallback);
+        }
+        BraveVpnNativeWorker.getInstance().getAllServerRegions();
+    }
+
+    @Override
+    public void onGetAllServerRegions(String jsonResponse, boolean isSuccess) {
+        if (isSuccess) {
+            BraveVpnPrefUtils.setBraveVpnServerRegions(jsonResponse);
+            new Handler().post(() -> updateSummaries());
+        } else {
+            Toast.makeText(getActivity(), R.string.fail_to_get_server_locations, Toast.LENGTH_LONG)
+                    .show();
+        }
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        if (BraveVpnUtils.mIsServerLocationChanged) {
+            BraveVpnUtils.mIsServerLocationChanged = false;
+            BraveVpnUtils.showProgressDialog(getActivity());
+            verifyPurchase(false);
+        } else {
+            BraveVpnUtils.dismissProgressDialog();
+        }
+        new Handler().post(() -> updateSummaries());
+    }
+
+    private void updateSummary(String preferenceString, String summary) {
+        Preference p = findPreference(preferenceString);
+        p.setSummary(summary);
+    }
+
+    private void updateSummaries() {
+        List<BraveVpnServerRegion> vpnServerRegions =
+                BraveVpnUtils.getServerLocations(BraveVpnPrefUtils.getBraveVpnServerRegions());
+        String serverLocation = "";
+        for (BraveVpnServerRegion vpnServerRegion : vpnServerRegions) {
+            if (BraveVpnPrefUtils.getServerRegion().equals(
+                        BraveVpnPrefUtils.PREF_BRAVE_VPN_AUTOMATIC)) {
+                serverLocation = getActivity().getResources().getString(R.string.automatic);
+            }
+            if (vpnServerRegion.getName().equals(BraveVpnPrefUtils.getServerRegion())) {
+                serverLocation = vpnServerRegion.getNamePretty();
+                break;
+            }
+        }
+        updateSummary(PREF_SERVER_LOCATION, serverLocation);
+        updateSummary(PREF_SERVER_HOST, BraveVpnPrefUtils.getBraveVpnHostname());
+        if (!BraveVpnPrefUtils.getBraveVpnProductId().isEmpty()) {
+            updateSummary(PREF_SUBSCRIPTION_STATUS,
+                    BraveVpnPrefUtils.getBraveVpnProductId().equals(
+                            InAppPurchaseWrapper.NIGHTLY_MONTHLY_SUBSCRIPTION)
+                            ? getActivity().getResources().getString(R.string.monthly_subscription)
+                            : getActivity().getResources().getString(R.string.yearly_subscription));
+        }
+
+        if (!BraveVpnPrefUtils.getBraveVpnPurchaseExpiry().isEmpty()) {
+            long expiresInMillis = Long.parseLong(BraveVpnPrefUtils.getBraveVpnPurchaseExpiry());
```

Comment:
Here also `parseLong` is used without exception handler. From other side, this value is from prefs and should be valid. 

Maybe put try/catch inside of `BraveVpnPrefUtils.getBraveVpnPurchaseExpiry` and in the case of failure give safe value and then check for it?

And also maybe for `BraveVpnUtils.getPurchaseExpiryDate` - return not string but parsed `long` with handling of possible exception? 

Besides this, there are several other places where `parseLong` is used in this PR.